en/simulation/multi-vehicle-simulation.md

```diff
@@ -39,88 +38,72 @@ To test an example setup, follow the below steps,
 
 For each simulated vehicle, the following is required,
 
-* **Gazebo model**: This is defined as `xacro` file in `Firmware/Tools/sitl_gazebo/models/rotors_description/urdf/<model>_base.xacro` see [here](https://github.com/PX4/sitl_gazebo/tree/02060a86652b736ca7dd945a524a8bf84eaf5a05/models/rotors_description/urdf). Currently, the model `xacro` file is assumed to end with base.xacro. This model should have an argument called  `mavlink_udp_port` which defines the UDP port on which gazebo will communicate with PX4 node. The model's `xacro` file will be used to generate an `sdf` model that contains UDP port that you select. To define the UDP port, set the `mavlink_udp_port` in the launch file, see [here](https://github.com/PX4/Firmware/blob/master/launch/multi_uav_mavros_sitl.launch#L48).  
+* **Gazebo model**: This is defined as `xacro` file in `Firmware/Tools/sitl_gazebo/models/rotors_description/urdf/<model>_base.xacro` see [here](https://github.com/PX4/sitl_gazebo/tree/02060a86652b736ca7dd945a524a8bf84eaf5a05/models/rotors_description/urdf). Currently, the model `xacro` file is assumed to end with base.xacro. This model should have an argument called  `mavlink_udp_port` which defines the UDP port on which gazebo will communicate with PX4 node. The model's `xacro` file will be used to generate an `urdf` model that contains UDP port that you select. To define the UDP port, set the `mavlink_udp_port` in the launch file for each vehicle, see [here](https://github.com/PX4/Firmware/blob/4d0964385b84dc91189f377aafb039d10850e5d6/launch/multi_uav_mavros_sitl.launch#L37) as an example.  
```

Comment:
- I would actually copy/paste a small fragment of the file as an example at the end. For the link, I'd perhaps link to master.
- "Currently, the model `xacro` file is assumed to end with base.xacro. " What is it that makes this assumption? Is it the launch file script?
- Assuming we use same file for all models, does that mean gazebo uses same port for all models to communicate with SITL? Presumably you could use different models if you wanted? In this case, are there any special rules?

Reply:
The `_base` is assumed [here](https://github.com/PX4/Firmware/blob/4d0964385b84dc91189f377aafb039d10850e5d6/launch/single_vehicle_spawn.launch#L19)

You need to have xacros for different vehicles, set the `vehicle` arg in the launch accordingly and then have rcS files for them. I imagine this (the iris model) came from the ETHz RotorS package -- they have more examples [here](https://github.com/ethz-asl/rotors_simulator/tree/master/rotors_description/urdf) within the PX4 repo this will only work with the iris. That is why I put the links on the bottom if users what to do something more specialized for their purpose.

The launch file defines the SITL port, which is passed to xacro and used to generate urdf model with the specified port. So, the ROS param server xacro generated urdf gazebo models communicate on different ports for gazebo <--> SITL, BUT the rcS needs to be made manually with a `SITL_UDP_PRT` param that matches this port number to work.

Rules -- I don't think so. Just make sure the ports are available on your system and won't overlap other applications or vehicles. We do this a bit differently in the lab. In order to make this less confusing we script all this port stuff and use a base like 14400, such that we have 144xy, where we increment x for each vehicle and y matches the digits used in the normal single vehicle launch.